freqtrade/rpc/api_server/api_ws.py

```diff
@@ -38,11 +42,26 @@ def _process_consumer_request(request: Dict[str, Any], channel: WebSocketChannel
             logger.debug(f"{channel} subscribed to topics: {data}")
             channel.set_subscriptions(data)
 
+    elif type == RPCRequestType.INITIAL_DATA:
+        # Acquire the data
+        initial_data = rpc._ws_initial_data()
+
+        # We now loop over it sending it in pieces
+        whitelist_data, analyzed_df = initial_data.get('whitelist'), initial_data.get('analyzed_df')
+
+        if whitelist_data:
+            await channel.send({"type": RPCMessageType.WHITELIST, "data": whitelist_data})
+
+        if analyzed_df:
+            for pair, message in analyzed_df.items():
+                await channel.send({"type": RPCMessageType.ANALYZED_DF, "data": message})
+
```

Comment:
That's not really different to implementing explicit handlers for `RPCRequestType.WHITELIST` and ` RPCRequestType.ANALYZED_DF` respectively - just that it's now force-feeding me 2 messages, which the consumer may not even need.

I'd split it up into 2 requests - so consumers can be explicit (while it may not be needed for the emc - it's a cleaner interface to respond to what is asked for explicitly).

If it's "initial data" - i'd assume i'll get "everything" - this will however depend on the consumer - emc is interested in these 2 types - but a UI may be interested in current open trades, historic trades, balances, but not in analyzed_df - so this would now suddenly be considered a bug or malfunction.
by calling something "Initial data" - there's no real argument why we should not send EVERYHING here.

Reply:
Like so? (eb4cd6b)